src/main/webapp/go.js

```diff
@@ -124,19 +174,43 @@ function hideStartButton() {
 }
 
 /**
- * Change the riddle text displayed on the mmain page.
- * @param {String} text Text that the riddle should be changed to.
+ * Update {code@ display} to specified {code@ text}.
+ * @param {String} display ID of element to be updated
+ * @param {String} text Text that display should be updated to.
+ */
+function updateMessage(display, text) {
+  const message = document.getElementById(display);
+  if (display == HINT_DISPLAY) {
```

Comment:
```suggestion
  if (display == HINT_DISPLAY) {
  // Special case for HINT_DISPLAY to avoid clearing out its existing contents because it may already contain previous hints.
```